src/main/kotlin/com/grepp/nbe1_3_team04/chat/service/ChatMemberServiceImpl.kt

```diff
@@ -0,0 +1,225 @@
+import com.grepp.nbe1_3_team04.chat.service.response.ChatMemberResponse
+import com.grepp.nbe1_3_team04.global.exception.ExceptionMessage
+import com.grepp.nbe1_3_team04.member.domain.Member
+import com.grepp.nbe1_3_team04.member.repository.MemberRepository
+import com.grepp.nbe1_3_team04.reservation.domain.Participant
+import com.grepp.nbe1_3_team04.team.domain.TeamMember
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+
+@Service
+class ChatMemberServiceImpl(
+    private val chatMemberRepository: ChatMemberRepository,
+    private val memberRepository: MemberRepository,
+    private val chatroomRepository: ChatroomRepository,
+    private val chatRepository: ChatRepository,
+    private val redisPublisher: RedisPublisher
+) : ChatMemberService {
+
+    /**
+     * 개인 채팅방 초대
+     *
+     * @param request
+     * @return
+     */
+    @Transactional
+    override fun joinChatMember(request: ChatMemberServiceRequest): ChatMemberResponse {
+        val member = getMember(request.memberId)
+        val chatroom = getChatroomByChatroomId(request.chatroomId)
+
+        return addChatMember(member, chatroom)
+    }
+
+    @Transactional
+    override fun joinTeamChatMember(member: Member, teamId: Long) {
+        val chatroom = getChatroomByTeamId(teamId)
+
+        addChatMember(member, chatroom)
+    }
+
+    @Transactional
+    override fun joinReservationChatMember(member: Member, reservationId: Long) {
+        val chatroom = getChatroomByReservationId(reservationId)
+
+        addChatMember(member, chatroom)
+    }
+
+    private fun addChatMember(member: Member, chatroom: Chatroom): ChatMemberResponse {
+        checkMemberNotInChatroom(member, chatroom)
+
+        val chatMember: ChatMember = chatMemberRepository.save(ChatMember.create(member, chatroom))
+
+        val chat = Chat.createEnterChat(chatroom, member)
+        chatRepository.save(chat)
+
+        redisPublisher.publish(chat)
+
+        return ChatMemberResponse(chatMember)
+    }
+
+    /**
+     * 팀원 채팅방 초대
+     *
+     * @param teamMembers
+     */
+    @Transactional
+    override fun joinChatTeam(teamMembers: List<TeamMember>, teamId: Long) {
+        val chatroom = getChatroomByTeamId(teamId)
+        val members = teamMembers.map { it.member }
+
+        joinChatMembers(members, chatroom)
+    }
+
+    /**
+     * 게임 참여 인원 채팅방 초대
+     *
+     * @param gameMembers
+     */
+    @Transactional
+    override fun joinChatGame(gameMembers: List<Participant>, reservationId: Long) {
+        val chatroom = getChatroomByReservationId(reservationId)
+
+        val members = gameMembers.map { it.member }
+
+        joinChatMembers(members, chatroom)
+    }
+
+    /**
+     * 단체로 채팅방 초대
+     *
+     * @param members
+     */
+    @Transactional
+    override fun joinChatMembers(members: List<Member>, chatroom: Chatroom) {
+        val oldMembersId = chatMemberRepository.findByChatroom(chatroom)
```

Comment:
```suggestion
        val oldMemberIds = chatMemberRepository.findByChatroom(chatroom)
```

좀 사소한 내용인데 List로 선언된 변수의 경우 아무래도 뒤에 s를 붙이는 게 더 좋아 보입니다! 영어를 잘 못하지만 표현적으로도 oldMemberIds 가 더 맞는 듯 하구요! 개인적인 의견입니다

Reply:
듣고보니 재경님 말씀이 맞습니다! 반영해보도록 할게요~